src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3178,6 +3370,66 @@ GetAggregateType(Aggref *aggregateExpression)
 		}
 	}
 
+	/*
+	 * All functions from github.com/tvondra/tdigest start with the "tdigest" prefix.
+	 * Since it requires lookups of function names in a schema we would like to only
+	 * perform these checks if there is some chance it will actually result in a positive
+	 * hit.
+	 */
+	if (StartsWith(aggregateProcName, "tdigest"))
+	{
```

Comment:
Given that this condition will only be true in very specific cases for queries that are low throughput / execution-heavy, do we really need OID caching for the TDigest functions? 

We mainly do caching because some functions OIDs are checked in practically every query (e.g. citus_extradata_container, read_intermediate_results), but we don't do it for HLL functions for instance.

What if we just do LookupFuncName in the TDigest functions and don't worry about cache invalidation? There's actually another layer of caching in PostgreSQL when looking up functions by name, so it will still be pretty fast.

It would also be nice to move the TDigest functions out of metadata_cache.c, which is already enormous.